FireboltNETSDK/FireboltClient.cs

```diff
@@ -152,19 +132,41 @@ public Task<GetEngineUrlByEngineNameResponse> GetEngineUrlByEngineId(string? eng
     /// </summary>
     /// <returns>A successful response.</returns>
     /// <exception cref="FireboltException">A server side error occurred.</exception>
-    public async Task<string?> ExecuteQueryAsync(string? engineEndpoint, string databaseName, string query,
-        HashSet<string> setParamList, CancellationToken cancellationToken)
+    public async Task<string?> ExecuteQueryAsync(string? engineEndpoint, string? databaseName, string? accountId,
+                         string query, HashSet<string> setParamList, CancellationToken cancellationToken)
     {
-        if (string.IsNullOrEmpty(engineEndpoint) || string.IsNullOrEmpty(databaseName) ||
-            string.IsNullOrEmpty(query))
+        if (string.IsNullOrEmpty(engineEndpoint) || string.IsNullOrEmpty(query))
             throw new FireboltException(
                 $"Some parameters are null or empty: engineEndpoint: {engineEndpoint}, databaseName: {databaseName} or query: {query}");
 
-        var setParam = setParamList.Aggregate(string.Empty, (current, item) => current + "&" + item);
-        var urlBuilder = new StringBuilder();
-        urlBuilder.Append("https://").Append(engineEndpoint).Append("?database=").Append(databaseName)
-            .Append(setParam).Append("&output_format=JSON_Compact");
-        return await SendAsync<string>(HttpMethod.Post, urlBuilder.ToString(), query, "text/plain", true, cancellationToken);
+        var setParams = setParamList.Aggregate(string.Empty, (current, item) => current + "&" + item);
+        var urlBuilder = new UriBuilder(engineEndpoint)
+        {
+            Scheme = "https",
+            Port = -1
```

Comment:
Do we need to provide a port here?

Reply:
We set port to -1, otherwise it will automatically add it to the resulting string, like `eng.uuid.us-east-1.dev.firebolt.io:443`, and that doesn't work in some cases.